libpod/runtime_img.go

```diff
@@ -268,12 +271,27 @@ func (r *Runtime) LoadImage(ctx context.Context, name, inputFile string, writer
 			newImages, err = r.ImageRuntime().LoadFromArchiveReference(ctx, src, signaturePolicy, writer)
 		}
 		if err != nil {
-			src, err := directory.NewReference(inputFile)
-			if err == nil {
-				newImages, err = r.ImageRuntime().LoadFromArchiveReference(ctx, src, signaturePolicy, writer)
+			if strings.Contains(name, "/") {
+				// prepend "localhost/" to support local image saved with this semantics
+				src, err = ociarchive.NewReference(inputFile, filepath.Join("localhost", name))
+				if err == nil {
+					newImages, err = r.ImageRuntime().LoadFromArchiveReference(ctx, src, signaturePolicy, writer)
+				}
 			}
 			if err != nil {
-				return "", errors.Wrapf(err, "error pulling %q", name)
+				src, err := directory.NewReference(inputFile)
+				if err == nil {
+					newImages, err = r.ImageRuntime().LoadFromArchiveReference(ctx, src, signaturePolicy, writer)
+				}
+				if err != nil {
+					src, err := layout.NewReference(inputFile, name)
```

Comment:
OCI expanded directories and OCI archives should behave the same WRT the prefixing with `localhost/`.